### PR TITLE
Revised our coding standards

### PR DESCRIPTION
**Pull request by JeroenVanSteijn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,51 +27,55 @@ Points that will be discussed in this Coding Standards file
...
-* Make sure your code is readable. This means using clear names for your functions, adding enough (although not too many) comments and adding enough and a consistent amount of whitespace
-* Always use linters when you are writing code, and let the linter check your code before committing on Git
-* When starting on a project, always pick the best tool for the job, we are not thinking inside of the box, but always discuss this with your superior
-* Always write your code with the latest techniques, and let tools like Babel or PostCSS take care or backwards compatibility
-* For the default folder structures, please see the end of this document
+* Structure your work in a way that other developers can easily understand it. This means having the possibility of co-workers modifying your code in the back of your mind
+* Document your code for end users, and when writing small and general functions
+* [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself): Don't repeat yourself! Don't build long functions that do a specific thing, but try to create smaller and more general functions
```
Comment (lucianonooijen) [inline README.md]: Can you add this to the `Sources` section?

File: README.md [hunk 2]
```diff
@@ -105,61 +111,59 @@ There are some general rules that should be followed at all times.
...
-* If a :hover pseudo class is styled, :focus should also be styled for accessibility
-* Use BEM for naming CSS classes
-* Name classes in a way it is clear to what is is about
-* For font-sizes, use an exponential scale, preferably based on the Fibonacci number (1.618)
-* For colors, use hex values (not rgb or rgba), as short as possible, and in lowercase, so don't use `#FFFFFF` but use `#fff` instead
+* If a :hover pseudo class is styled, style :focus the same for accessibility
+* Use [BEM](http://getbem.com/naming/) for naming CSS classes
+* Name classes descriptively 
+* For font-sizes, use an exponential scale, preferably based on the Fibonacci number (1.618) [Click here to geek out](https://3.7designs.co/blog/2010/10/how-to-design-using-the-fibonacci-sequence/)
+* For colors, use hex values (not rgb or rgba), as short as possible, and in lowercase, for example: not `#FFFFFF` but `#fff`
```
Comment (lucianonooijen) [inline README.md]: I would like to add that we use `background: #000` and not `background-color: #000`

File: README.md [hunk 3]
```diff
@@ -105,61 +111,59 @@ There are some general rules that should be followed at all times.
...
-* If you are writing emails, use the MJML preprocessor
-* For production, remove source maps and minify the code
+* If you are writing emails, use of the MJML preprocessor is encouraged
 
 # CSS preprocessors
 
 * Currently, SCSS is preferred for a project. If you want to use something else, discuss this with your superior.
 * Make use of the functionalities the preprocessor gives you! Don't do stuff yourself that a preprocessor can take care off for you
-* You can compile your code to CSS using Webpack or Gulp
+* You can compile your code to CSS using toolkits like Webpack or Gulp
```
Comment (lucianonooijen) [inline README.md]: Can you make it `Webpack, Gulp or Parcel`

File: README.md [hunk 4]
```diff
@@ -105,61 +111,59 @@ There are some general rules that should be followed at all times.
...
 * Don't go and add prefixes by hand. Instead, let a plugin handle this for you
-* Try to use nesting at least as possible, rather use BEM. Use the nesting mostly for :hover-states etc.
-* Only nest 3 levels deep
+* Try to avoid nesting, rather use BEM. Use of nesting is mostly resticted to :hover-states etc.
+* When you do nest, make it at most 3 levels deep
 * Break files out into small modules (avoid having a file that is larger than 100 lines)
 * Avoid using IDs throughout the site. Use IDs for parent elements. Example: Header, Footer, Main. Using Classes avoids having to use !important
-* For shameful code, use the _shame.scss file
+* For shameful code, use the *_shame.scss file*. We all have to do this sometimes, better own up to it. [More Geeky links](https://csswizardry.com/2013/04/shame-css/)
```
Comment (lucianonooijen) [inline README.md]: Can you add this to the `Sources` section?

File: README.md [hunk 5]
```diff
@@ -105,61 +111,59 @@ There are some general rules that should be followed at all times.
...
 * Use the folder structure as given in this project
 
 # Javascript
 
-* If possible, use vanilla Javascript, so don't add jQuery just for a simple DOM manipulation or a simple API call
-* Use Yarn as a package manager, not NPM
+* If possible, use vanilla Javascript. Don't add jQuery just for a simple DOM manipulation or a simple API call
+* Use Yarn as a package manager: [it's better than NPM](https://www.sitepoint.com/yarn-vs-npm/).
```
Comment (lucianonooijen) [inline README.md]: Can you add this to the `Sources` section?

Comment (lucianonooijen) [reply README.md]: Maybe we can make subsections for the sources?

File: README.md [hunk 6]
```diff
@@ -105,61 +111,59 @@ There are some general rules that should be followed at all times.
...
 * The usage of Typescript is encouraged
-* For linting, use ESLint
+* For linting, use ESLint. You're allowed to use modules on top of ESLint like [Prettier](https://prettier.io/)
 * Write your code in ESNext/ES6+, and transpile using Babel to ES5 because Internet Explorer
-* Before adding a library, discuss this with your superior
+* Before adding a library, discuss with your superior
 * Don't use anonymous functions (`function test() {}`), write them like this: `const test = () => {}`
-* When using jQuery, write it using `$('selector');` and not using `jQuery('selector');`
-* For production, uglify/minify your code and remove source maps
```
Comment (lucianonooijen) [inline README.md]: Why was this line removed? Can you add this again, maybe to the `General` section? And mention it is for CSS and JS

File: README.md [hunk 7]
```diff
@@ -105,61 +111,59 @@ There are some general rules that should be followed at all times.
...
+* If you are using another framework that uses px-units, use px, else use rem-values *(rem values scale with the HTML-object's font-size, which makes responsive typography a whole lot easier)*
+* There should only be a maximum of 2 CSS files per page (to decrease HTTP requests)
 * List related items together
-* If a :hover pseudo class is styled, :focus should also be styled for accessibility
-* Use BEM for naming CSS classes
-* Name classes in a way it is clear to what is is about
-* For font-sizes, use an exponential scale, preferably based on the Fibonacci number (1.618)
-* For colors, use hex values (not rgb or rgba), as short as possible, and in lowercase, so don't use `#FFFFFF` but use `#fff` instead
+* If a :hover pseudo class is styled, style :focus the same for accessibility
+* Use [BEM](http://getbem.com/naming/) for naming CSS classes
```
Comment (lucianonooijen) [inline README.md]: Can you add this to the `Sources` section?

File: README.md [hunk 8]
```diff
@@ -105,61 +111,59 @@ There are some general rules that should be followed at all times.
...
+* Name classes descriptively 
+* For font-sizes, use an exponential scale, preferably based on the Fibonacci number (1.618) [Click here to geek out](https://3.7designs.co/blog/2010/10/how-to-design-using-the-fibonacci-sequence/)
```
Comment (lucianonooijen) [inline README.md]: Can you add this to the `Sources` section?

File: README.md (hunk 6 above)
Comment (JeroenVanSteijn) [reply README.md]: Right, I think that was the point (DRY ;) ), but I can't find it in general either. Will add this back to general.

**Comment by JeroenVanSteijn** [comment]
@lucianonooijen changes were made.

File: README.md [hunk 9]
```diff
@@ -105,61 +112,59 @@ There are some general rules that should be followed at all times.
...
+* For colors, use hex values (not rgb or rgba), as short as possible, and in lowercase, for example: not `#FFFFFF` but `#fff`, and not `background-color` but ''
```
Comment (lucianonooijen) [inline README.md]: I think there's something missing in here 😄
